pkg/util/extract_forwarded.go

```diff
@@ -0,0 +1,91 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+
+	"google.golang.org/grpc/metadata"
+)
+
+// ipAddressesKey is key for the GRPC metadata where the IP addresses are stored
+const ipAddressesKey = "ipaddresseskey"
+
+// extractHost returns the Host IP address without any port information
+func extractHost(address string) string {
+	hostIP := net.ParseIP(address)
+	if hostIP != nil {
+		return hostIP.String()
+	}
+	var err error
+	hostStr, _, err := net.SplitHostPort(address)
+	if err != nil {
+		// Invalid IP address, just return it so it shows up in the logs
+		return address
+	}
+	return hostStr
+}
+
+// GetSource extracts the X-FORWARDED-FOR header from the given HTTP request
+// and returns a string with it and the remote address
+func GetSource(req *http.Request) string {
+	fwd := req.Header.Get("X-FORWARDED-FOR")
```

Comment:
This code has move to weaveworks/common. I'll add it there.

Reply:
I'm not too keen on restricting the X-Forwarded-For headers just yet. Customers haven't used this feature yet so there will be mistakes. By removing wrong addresses it will make it harder to see what was used.